crates/subspace-service/src/sync_from_dsn.rs

```diff
@@ -304,14 +304,16 @@ where
         .map_err(|error| error.to_string())?;
 
     // Corresponds to contents of block one, everyone has it, so we consider it being processed
-    // right away
+    // right away.
+    // TODO: implement starting from any segment index, and handle the partial block at the end of
+    // the last segment.
     let mut last_processed_segment_index = SegmentIndex::ZERO;
     // TODO: We'll be able to just take finalized block once we are able to decouple pruning from
     //  finality: https://github.com/paritytech/polkadot-sdk/issues/1570
     let mut last_processed_block_number = info.best_number;
     let segment_header_downloader = SegmentHeaderDownloader::new(node);
 
-    while let Some(reason) = notifications.next().await {
+    if let Some(reason) = notifications.next().await {
```

Comment:
I don't think this is a correct change, can you explain?

Right now this whole commit doens't make a lot of sense and I don't understand the problem it tries to fix. Except that it is clear that it breaks DSN sync right after its initial completion and will likely result in immediate node shutdown due to exit of essential task (I have not tried to run it though, did you?).